src/librustc_mir/transform/promote_consts.rs

```diff
@@ -266,8 +277,10 @@ impl<'a, 'tcx> Promoter<'a, 'tcx> {
         });
 
         // Inject the Rvalue or Call into the promoted MIR.
+        let dest = Lvalue::Temp(new_temp);
+        self.push(StatementKind::StorageLive(dest.clone()), source_info.span);
```

Comment:
Sure, `&&&5` in a constant will create 3 temporaries in a sort of "borrow chain".